pkg/legacy/tarball/write_test.go

```diff
@@ -320,6 +320,48 @@ func TestWriteForeignLayers(t *testing.T) {
 	}
 }
 
+func TestMultiWriteNoHistory(t *testing.T) {
```

Comment:
Since you added code to handle history and layers >0 but history != layers, is there a test which handles that? Or should a test be added for that?